File: Data Structures & Algorithms/binary-search/submission-4.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0; 
        int r = nums.length - 1;
        int midPoint = (l + r) / 2;
        while (l <= r) {
            midPoint = (l + r) / 2;
            if (nums[midPoint] == target) {
                return midPoint;
            }

            else if (target < nums[midPoint]) {
                r = midPoint - 1;
            }

            else {
                l = midPoint + 1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zeroCount = 0
        product = 1 
        for i in nums :
            if i == 0 :
                zeroCount += 1 
                if zeroCount == 1 :
                    continue 
                elif zeroCount >= 2 :
                    return [0 for i in range(len(nums))]
            product *= i

        res = []
        if zeroCount == 1 :
            for i in nums :
                if i == 0 :
                    res.append(product)
                else :
                    res.append(0)
        
        else :
            for i in nums :
                res.append(product//i)

        return res
        
        